chatsky/context_storages/database.py

```diff
@@ -8,181 +8,135 @@
 This class implements the basic functionality and can be extended to add additional features as needed.
 """
 
-import asyncio
-import importlib
-import threading
-from functools import wraps
+import pickle
 from abc import ABC, abstractmethod
-from typing import Callable, Hashable, Optional
+from importlib import import_module
+from typing import Any, Hashable, List, Literal, Optional, Set, Tuple, Union
+
+from pydantic import BaseModel, Field
 
 from .protocol import PROTOCOLS
-from ..script import Context
 
 
-class DBContextStorage(ABC):
-    r"""
-    An abstract interface for `chatsky` DB context storages.
-    It includes the most essential methods of the python `dict` class.
-    Can not be instantiated.
-
-    :param path: Parameter `path` should be set with the URI of the database.
-        It includes a prefix and the required connection credentials.
-        Example: postgresql+asyncpg://user:password@host:port/database
-        In the case of classes that save data to hard drive instead of external databases
-        you need to specify the location of the file, like you do in sqlite.
-        Keep in mind that in Windows you will have to use double backslashes '\\'
-        instead of forward slashes '/' when defining the file path.
+class FieldConfig(BaseModel, validate_assignment=True):
+    """
+    Schema for :py:class:`~.Context` fields that are dictionaries with numeric keys fields.
+    Used for controlling read and write policy of the particular field.
+    """
 
+    name: str = Field(default_factory=str, frozen=True)
```

Comment:
I think default value should be `turns`.

Reply:
Yes, but I suppose the most common use case for initializing `FieldConfig` is to change the number of loaded turns.
Also, why is it an empty string by default?
Why allow arbitrary names instead of `Literal["turns", "misc"]`?